gdk4/src/texture.rs

```diff
@@ -14,6 +15,18 @@ pub trait TextureExtManual: 'static {
 impl<O: IsA<Texture>> TextureExtManual for O {
     fn download(&self, data: &mut [u8], stride: usize) {
         unsafe {
+            assert!(
+                stride >= 4,
+                "Stride for a CAIRO_FORMAT_ARGB32 should be >= 4"
+            );
+            assert!(
+                data.len() as i32 >= 4 * (stride as i32) * self.as_ref().height(),
+                "The data is not big enough to download the texture"
+            );
+            assert!(
```

Comment:
And you probably want to switch the order of the second and third assertion for logical reasons